### PR TITLE
Add tutorial for state histograms

### PR DESCRIPTION
**Pull request by tanujkhattar**
This PR adds a tutorial for using the new and improved `cirq.vis.plot_state_histogram` explaining the various ways of processing the results and configuring the plots. 
This is part of the larger roadmap item #3642

### CODE REVIEW

File: docs/tutorials/state_histograms.ipynb
Comment (95-martin-orion) [inline docs/tutorials/state_histograms.ipynb:1]: <p>Question: what is the <code>plt.subplots()[1]</code>  doing here? Could we just use the default axes instead?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/quantumlib/Cirq/pull/4024/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (95-martin-orion) [inline docs/tutorials/state_histograms.ipynb:1]: <p>Recommend "For example" rather than "For eg"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/quantumlib/Cirq/pull/4024/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (95-martin-orion) [inline docs/tutorials/state_histograms.ipynb:1]: <p><code>Pass it's result</code>  -&gt; <code>Pass its result</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/quantumlib/Cirq/pull/4024/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (95-martin-orion) [inline docs/tutorials/state_histograms.ipynb:1]: <p>No need to repeat the simulation here, unless you feel that it adds clarity.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/quantumlib/Cirq/pull/4024/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (95-martin-orion) [inline docs/tutorials/state_histograms.ipynb:1]: <p><code>used process</code>  -&gt; <code>used to process</code></p><p> <code>genrate</code>  -&gt; <code>generate</code></p><p><code>of parity of number</code> -&gt; <code>of the parity of the number</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/quantumlib/Cirq/pull/4024/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (tanujkhattar) [reply docs/tutorials/state_histograms.ipynb:1]: It will create a new subplot and pass the axes to the function. If we don't pass a new axes, the `plot_state_histogram` will create a new one itself but then it'll also implicitly call `fig.show()` which leads to an ugly warning regarding the back-end being a non-GUI backend. 

Tl;Dr : This is needed to ensure there are no unnecessary warnings in the notebook.

Comment (tanujkhattar) [reply docs/tutorials/state_histograms.ipynb:1]: done.

Comment (tanujkhattar) [reply docs/tutorials/state_histograms.ipynb:1]: done

Comment (tanujkhattar) [reply docs/tutorials/state_histograms.ipynb:1]: removed the repetition.

Comment (tanujkhattar) [reply docs/tutorials/state_histograms.ipynb:1]: done

**Review by 95-martin-orion — APPROVED** [review]
Good to go once the merge conflict is resolved.

**Pull request merged**
